index.html

```diff
@@ -1076,11 +1113,11 @@ <h2>Fragment</h2>
   document</a> to identify a component of the document (for example, a unique
   <a>public key description</a> or <a>service endpoint</a>). <a>DID fragment</a>
   syntax and semantics are identical to a generic URI fragment and  MUST conform
-  to <a data-cite="RFC3986#section-3.5">RFC&nbsp;3986, section 3.5</a>. To resolve
+  to <a data-cite="RFC3986#section-3.5">RFC&nbsp;3986, section 3.5</a>. To dereference
   a <a>DID fragment</a> reference, the complete <a>DID URL</a> including the
-  <a>DID fragment</a> MUST be used as input to the <a>DID URL</a> dereferencing
-  algorithm for the target component in the <a>DID document</a> object. For more
-  information, see [[?DID-RESOLUTION]].
+  <a>DID fragment</a> MUST be used as input to the <a>DID URL dereferencing</a>
```

Comment:
just catching this now. Would it be considered derefencing if I used `did:example:123?id=key1`? I know we still have to discuss the usage of id query parameter, but figured it should be considered here as well.

Reply:
> I noticed the contract has been removed at this point, but I'm hoping it will come back with your addition of normative statements.

Yes, that's planned in the next PR that layers on top of this one.

Reply:
> Would it be considered derefencing if I used did:example:123?id=key1?

Yes, given that id=key1 becomes a DID Parameter whose purpose is to dereference to a key in the DID Document.

That said, I think that doing this is a bad idea. :)

Reply:
> 
> 
> > Would it be considered derefencing if I used did:example:123?id=key1?
> 
> Yes, given that id=key1 becomes a DID Parameter whose purpose is to dereference to a key in the DID Document.
> 
> That said, I think that doing this is a bad idea. :)

Would doing that require an extended document loader to handle that de-referencing?

Reply:
It's the concept of "get me this thing that is expressed in the DID Document using this syntax that duplicates the use of fragment identifiers" that is problematic. If you want the key, get the DID Document, look for the key's fragment identifier, and you're done... w/o needing a DID Parameter for that feature. It can be done w/ a simple convenience function.

Reply:
could `did:example:123?id=key1` be used as an iri as well if the producer chose to do so? Really just trying to understand the edge cases here. I stick to fragments here normally cause they're easier to deal with and commonly used by everyone.